app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -199,44 +198,16 @@ class EventRepositoryImpl @Inject constructor(
         }
     }
 
-    private data class CompressEventResult(val event: AgendaItem.Event, val photosDeleted: Int)
-    private suspend fun compressEvent(event: AgendaItem.Event): CompressEventResult {
-        return withContext(Dispatchers.Default) {
-            var photosDeleted = 0
-            val compressedEvent = event.copy(
-                photos = supervisorScope {
-                    event.photos
-                        .chunked(MAX_PARALLEL_IMAGE_COMPRESS_COUNT)
-                        .map { eventPhotoChunk ->
-                            eventPhotoChunk.map { eventPhoto ->
-                                async {
-                                    if (eventPhoto is LocalPhoto) {
-                                        if (eventPhoto.savedInternally) {
-                                            val byteArray =
-                                                imageStorage.getByteArray(eventPhoto.key)
-                                            eventPhoto.copy(byteArray = byteArray)
-                                        } else {
-                                            val byteArray = eventPhoto.bitmap?.let { bitmap ->
-                                                bitmapConverter.bitmapToCompressByteArray(
-                                                    bitmap, MAX_PHOTO_SIZE_IN_BYTES
-                                                )
-                                            }
-                                            if (byteArray != null) {
-                                                eventPhoto.copy(byteArray = byteArray)
-                                            } else {
-                                                photosDeleted++
-                                                null
-                                            }
-                                        }
-                                    } else {
-                                        eventPhoto
-                                    }
-                                }
-                            }.mapNotNull { it.await() }
-                        }.flatten()
+    override suspend fun getLocalPhotos(localPhotoKeys: List<String>): List<EventPhoto.Local> {
+        return withContext(Dispatchers.IO) {
+            supervisorScope {
+                async {
+                    localPhotoKeys.map { key ->
+                        val bitmap = imageStorage.getBitmap(key)
+                        EventPhoto.Local(key = key, bitmap = bitmap, savedInternally = true)
+                    }
                 }
```

Comment:
Wrong place for async, this will execute the whole mapping inside async and therefore still load the bitmaps sequentially.
Instead, the async block should be inside the map block so an independent coroutine is launched for every call to `getBitmap()`